src/code/z_camera.c

```diff
@@ -55,31 +256,184 @@ s32 Camera_IsSwimming(Camera* camera) {
+
+        if (!ret) {
+
+            ret = false;
+
+            if (camera->focalActor->gravity > -0.1f) {
+                ret = true;
+            }
+
+            player = (Player*)camera->focalActor;
+            if (!ret) {
+                // Using zora fins
+                ret = player->stateFlags1 & PLAYER_STATE1_200000;
+                ret = !!ret;
+            }
+        }
+        return ret;
+    } else {
+        return true;
+    }
+}
+
+s32 Camera_IsClimbingLedge(Camera* camera) {
+    Actor* focalActor = camera->focalActor;
+
+    if (focalActor == &GET_PLAYER(camera->play)->actor) {
+        return ((Player*)focalActor)->stateFlags1 & PLAYER_STATE1_4;
+    } else {
+        return 0;
+    }
+}
+
+s32 Camera_IsChargingSwordOrDekuFlowerDive(Camera* camera) {
+    Actor* focalActor = camera->focalActor;
+    s32 ret;
+
+    if (focalActor == &GET_PLAYER(camera->play)->actor) {
+        // Charging Sword
+        ret = !!(((Player*)focalActor)->stateFlags1 & PLAYER_STATE1_1000);
+        if (!ret) {
+            // Deku Flower Dive
+            ret = !!(((Player*)focalActor)->stateFlags3 & PLAYER_STATE3_100);
+        }
+        return ret;
+    } else {
+        return false;
+    }
+}
+
+s32 func_800CBA7C(Camera* camera) {
+    Actor* focalActor = camera->focalActor;
+
+    if (camera->focalActor == &GET_PLAYER(camera->play)->actor) {
+        return ((Player*)focalActor)->stateFlags2 & PLAYER_STATE2_800000;
+    } else {
+        return 0;
+    }
+}
+
+s32 func_800CBAAC(Camera* camera) {
+    Actor* focalActor = camera->focalActor;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CB8F4.s")
+    if (camera->focalActor == &GET_PLAYER(camera->play)->actor) {
+        return ((Player*)focalActor)->meleeWeaponState;
+    } else {
+        return 0;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CB924.s")
+s32 func_800CBAD4(Vec3f* dst, Camera* camera) {
+    PosRot focalPosRot;
+    Actor* focalActor = camera->focalActor;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CB950.s")
+    if (camera->focalActor == &GET_PLAYER(camera->play)->actor) {
+        *dst = ((Player*)focalActor)->bodyPartsPos[0];
+        return dst;
+    } else {
+        Actor_GetWorldPosShapeRot(&focalPosRot, camera->focalActor);
+        *dst = focalPosRot.pos;
+        return dst;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CBA08.s")
+s32 Camera_IsUnderwaterAsZora(Camera* camera) {
+    Actor* focalActor = camera->focalActor;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CBA34.s")
+    if (camera->focalActor == &GET_PLAYER(camera->play)->actor) {
+        return ((Player*)focalActor)->currentBoots == PLAYER_BOOTS_ZORA_UNDERWATER;
+    } else {
+        return 0;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/func_800CBA7C.s")
+/**
+ * Evaluate if player is in one of two sword animations
+ */
+s32 func_800CBB88(Camera* camera) {
```

Comment:
Are these the ones that wobble the camera a bit?

Reply:
Yes actually. This function has only 1 use-case and it's to apply distortion/wobble:
https://github.com/engineer124/mm/blob/a4b03a08db5e35c912e4bd0f829abbeb590c1761/src/code/z_camera.c#L7315-L7341

Reply:
Mm, looks like case 1 is mislabeled in that file? Possibly this should get an enum later, not much point for now.